src/uu/csplit/src/csplit.rs

```diff
@@ -379,23 +406,33 @@ impl SplitWriter<'_> {
             input_iter.set_size_of_buffer(1);
 
             while let Some((ln, line)) = input_iter.next() {
-                let l = line?;
+                let line = line?;
+                let l = if let Some(l) = line.strip_suffix('\n') {
+                    if let Some(l) = l.strip_suffix('\r') {
+                        l
+                    } else {
+                        l
+                    }
+                } else {
+                    &line
+                }
+                .to_string();
```

Comment:
While this works fine, it's a bit odd if the `if` and `else` cases look the same ;-)

I think you can simplify it to something like:
```suggestion
                let l = line
                    .strip_suffix("\r\n")
                    .unwrap_or_else(|| line.strip_suffix('\n').unwrap_or(&line));
```